### ISSUE TITLE
The `identifier_format` optarg does not work.

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#5986]
When specifying `identifier_format='uuid'`, `r.table()` will still work given the name of a table rather than a UUID.  If you pass a UUID, the request will error with, e.g.,  `Table name '20bf6257-41c5-4a2f-b836-42d90c3fe321' invalid (Use A-Za-z0-9_ only)`.

Working on this now, as I'm blocked on Horizon stuff until this is fixed.

**Comment by Tryneus** [comment]
Apparently I had a misconception about how `indentifier_format` applies to `r.table()`.  Opened a new issue (#5987) for adding the optarg I actually wanted.
